common/src/main/java/com/revenuecat/purchases/common/offlineentitlements/OfflineEntitlementsManager.kt

```diff
@@ -78,18 +78,22 @@ class OfflineEntitlementsManager(
         )
     }
 
-    fun updateProductEntitlementMappingCacheIfStale() {
+    fun updateProductEntitlementMappingCacheIfStale(completion: (() -> Unit)? = null) {
```

Comment:
Added a completion here to be able to wait upon it on tests. It won't be used in production. Still it's tested.

Reply:
As an improvement you could forward the error too so integration tests can ensure it actually worked?